Editor - integrating basic visual editor

aforcier opened the pull request:
This PR integrates an early version of the new hybrid editor into WPAndroid for testing only.

![wpandroid-visual-editor-jun-2015](https://cloud.githubusercontent.com/assets/9613966/8185463/94c1dd3c-1412-11e5-90b6-b038c6a381ef.png)

Things that should be working:

* Creating a post/page
* Editing a post/page
* Saving draft, auto-save
* Publishing a post/page
* Basic formatting (bold, italics, strikethrough (tablet only), lists, blockquotes)

Not yet implemented:

* Adding media and links
* HTML editing mode
* Preview (existing preview still being used)
* Post settings

There are a few known bugs listed [here](https://github.com/wordpress-mobile/WordPress-Editor-Android/labels/bug).

aforcier commented:
Added a toggle for the new editor in the settings screen:

![editor-settings-toggle](https://cloud.githubusercontent.com/assets/9613966/8236180/a5ef8bb0-15b4-11e5-83ef-4d839fd9968e.png)

maxme commented on WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java:
```diff
@@ -167,6 +173,12 @@ protected void onCreate(Bundle savedInstanceState) {
         setContentView(R.layout.new_edit_post_activity);
 
+        // TODO: Remove debug build check when new editor is moved to production
+        if (PackageUtils.isDebugBuild()) {
```
We should remove the `isDebugBuild` check here or hide the option in settings when not in a debug build. 

IMO, it's safe to remove it since we're not going to merge `feature/visual-editor` in `develop` until the editor is ready.

maxme commented on WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java:
```diff
@@ -450,7 +462,7 @@ public boolean onOptionsItemSelected(MenuItem item) {
             }
 
-            savePost(false, false);
+            savePostToDb();
```
:+1:

bummytime commented:
@aforcier I noticed a small bug. 

When you first install the app the visual editor is enabled in the account settings, however when you edit a post, the old editor remains. If I go into account settings and toggle the editor off and then on again, the  `pref_key_visual_editor_enabled` will be set to `true` and everything works just fine.

aforcier commented:
* Dropped the `isDebugBuild` check
* Fixed the issue where the old editor would be displayed until you visited the settings activity

Ready for another look @maxme, @bummytime!

maxme commented:
:shipit:
